File: app/src/main/java/com/egecius/rxjava2_demo_2/rx/subjects/MaybeSubjectExamples.java
package com.egecius.rxjava2_demo_2.rx.subjects;

import com.egecius.rxjava2_demo_2.rx.EgisException;

import io.reactivex.subjects.MaybeSubject;

public class MaybeSubjectExamples {

    private final MaybeSubject<String> mMaybeSubject = MaybeSubject.create();

    public MaybeSubject<String> getMaybeSubject() {
        return mMaybeSubject;
    }

    void emitSingleError() {
        mMaybeSubject.onError(new EgisException());
    }
}

File: app/src/main/java/com/egecius/rxjava2_demo_2/rx/subjects/MaybeSubjectExamples.kt
package com.egecius.rxjava2_demo_2.rx.subjects

import com.egecius.rxjava2_demo_2.rx.EgisException

import io.reactivex.subjects.MaybeSubject

class MaybeSubjectExamples {

    val maybeSubject = MaybeSubject.create<String>()

    fun emitSingleError() {
        maybeSubject.onError(EgisException())
    }

    fun emit2Errors() {
        maybeSubject.onError(EgisException())
        maybeSubject.onError(EgisException())
    }
}
